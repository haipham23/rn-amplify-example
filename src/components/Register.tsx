import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import {
  Item,
  Form,
  Input,
  Button,
  Text
} from 'native-base';

interface Props {
  username: string,
  email: string,
  password: string,
  updateUsername(username: string): void,
  updateEmail(emai: string): void,
  updatePassword(password: string): void,
  signUp(): void,
}

const Register = ({
  username,
  email,
  password,
  updateUsername,
  updateEmail,
  updatePassword,
  signUp
}: Props) => (
  <View style={styles.content}>
    <Form>
      <Item floatingLabel>
        <Input
          placeholder="Username"
          value={username}
          onChangeText={updateUsername}
        />
      </Item>
      <Item floatingLabel>
        <Input
          placeholder="Email"
          value={email}
          onChangeText={updateEmail}
        />
      </Item>
      <Item floatingLabel last>
        <Input
          secureTextEntry
          placeholder="Password"
          value={password}
          onChangeText={updatePassword}
        />
      </Item>
      <Button full style={styles.button} onPress={signUp}>
        <Text>Sign Up</Text>
      </Button>
    </Form>
  </View>
);

const styles = StyleSheet.create({
  button: {
    marginVertical: 40
  },
  content: {
    padding: 10
  }
});

export default Register;
