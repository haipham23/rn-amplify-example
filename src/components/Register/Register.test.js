import * as React from 'react';
import renderer from 'react-test-renderer';

import Register from './Register';

describe('<Register />', () => {
  const updateUsername = jest.fn();
  const updateEmail = jest.fn();
  const updatePassword = jest.fn();
  const signUp = jest.fn();

  it('should render', () => {
    const tree = renderer.create(
      <Register
        username="testUser"
        email="test@example.com"
        password="12345678"
        updateUsername={updateUsername}
        updateEmail={updateEmail}
        updatePassword={updatePassword}
        signUp={signUp}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
