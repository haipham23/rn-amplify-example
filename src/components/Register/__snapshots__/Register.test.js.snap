// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      focusable={false}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "paddingTop": 5,
              "top": 6,
            }
          }
        >
          <Text
            float={true}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontSize": 15,
                  "lineHeight": 30,
                },
              ]
            }
          />
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Username"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "marginLeft": null,
            "paddingBottom": 7,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 3,
            "top": 10,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="testUser"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 15,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "left": 0,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "paddingTop": 5,
              "top": 6,
            }
          }
        >
          <Text
            float={true}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontSize": 15,
                  "lineHeight": 30,
                },
              ]
            }
          />
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Email"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "marginLeft": null,
            "paddingBottom": 7,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 3,
            "top": 10,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="test@example.com"
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginLeft": 0,
          "marginTop": 15,
          "opacity": 1,
          "paddingLeft": 15,
        }
      }
    >
      <View
        style={
          Object {
            "left": 15,
            "opacity": 1,
            "paddingBottom": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 0,
            "top": 18,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#575757",
              "fontSize": 17,
              "left": 0,
              "paddingRight": 5,
              "paddingTop": 5,
              "top": 6,
            }
          }
        >
          <Text
            float={true}
            style={
              Array [
                Object {
                  "fontSize": 17,
                },
                Object {
                  "fontSize": 15,
                  "lineHeight": 30,
                },
              ]
            }
          />
        </Text>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Password"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "marginLeft": null,
            "paddingBottom": 7,
            "paddingLeft": null,
            "paddingRight": 5,
            "paddingTop": 3,
            "top": 10,
          }
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value="12345678"
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 0,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "marginVertical": 40,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Sign Up
      </Text>
    </View>
  </View>
</View>
`;
